### PR TITLE
Typescript *.d.ts and typings.d.ts generation

### PR DESCRIPTION
**Pull request by lordnox**
As mentioned in #7, this pull request updates the `generate.js` to also create a `.d.ts` file for each icon file. It will go on and generate an `index.js`as well as the `typings.d.ts`

I did not change the version yet and you might want to remove the added dependencies of react & @typings/react as they are not actually needed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,5 +23,9 @@
     "babel-preset-es2015": "^6.9.0",
     "babel-preset-react": "^6.11.1",
     "rimraf": "^2.6.1"
+  },
+  "dependencies": {
+    "@types/react": "^16.0.38",
+    "react": "^16.2.0"
```
Comment (levrik) [inline package.json]: Does this need to be in the `dependencies`?

Comment (levrik) [reply package.json]: Ah. I should have read the last paragraph 😄

File: scripts/generate.js [hunk 2]
```diff
@@ -27,8 +43,38 @@ const ${name}Icon = ({ width = 24, height = 24, viewBox = '0 0 24 24', className
...
+
+${icons.map(nameIcon =>
+`declare const ${nameIcon}: React.ComponentType<ReactMdiIconProps>;
+`).join('')}
+`;
+fs.writeFileSync(Path.join(publishPath, 'typings.d.ts'), typingsContent);
+
+// Build the index.js file, before babel compile
+const indexContent = icons.map(nameIcon => `export ${nameIcon} from './${nameIcon}'`).join('\n')
+fs.writeFileSync(Path.join(buildPath, 'index.js'), indexContent);
```
Comment (levrik) [inline scripts/generate.js]: Personally I'm against having this `index.js`.
It was already there and I removed it for a reason. It's just too easy to include all icons in your browser bundle this way by accident.

Comment (lordnox) [reply scripts/generate.js]: I know what you mean, before there was webpack I'd said the same thing. But now we do have dead code elimination and tree shaking almost as the default.
There are even rules to support this: https://palantir.github.io/tslint/rules/no-submodule-imports/

The rule is actually the reason I did include it...

Libraries like lodash do this as well. If you want I'll remove it, but I would like it in there.

**Comment by lordnox** [comment]
Please review again; I removed the index.js and the typings for it. 
Additionally I updated the .gitignore as it likes to add .DS_Store files... 
And, of course, I moved the dependencies for react to peerDependency

File: scripts/generate.js (hunk 2 above)
Comment (levrik) [reply scripts/generate.js]: Let's merge it in without the `index.js` for now. I will investigate again if it can be added back without issues 🙂

Comment (lordnox) [reply scripts/generate.js]: Yeah, the last version does not have the index.js (and typings for it) in it. I pushed this about 2hours ago

**Comment by levrik** [comment]
@lordnox Can you check if your PR is compatible with the changes in #15?
You could also re-add the definitions for `index.js`.

**Comment by lordnox** [comment]
Pew... There were quite some conflicts.
This should be working, I tested it with a small .tsx file, and the typings worked fine.  I cannot say anything about the index.* things you merged in #15.

I also removed the children from the typings, this means this will not work anymore:
```
import SomeIcon from 'mdi-react';
// in typescript:
// import * as SomeIcon from 'mdi-react';
// this works:
import { OtherIcon } from 'mdi-react';

// this will throw a type error, as children are forbidden
const noRender = () => (
  <SomeIcon>Things that are not shown<SomeIcon>
)

const renderThatWorks = () => (
  <SomeIcon><SomeIcon>
)

// this should be the one to use
const render = () => (
  <SomeIcon />
)
```

**Comment by levrik** [comment]
Merged with a few changes in a86ab93cb18456662b70767828f31730419392ea

**Pull request closed without merging**